v03_pipeline/lib/misc/family_loading_failures.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from v03_pipeline.lib.misc.pedigree import Relation
+from v03_pipeline.lib.model import Ploidy
+
+if TYPE_CHECKING:
+    import hail as hl
+
+    from v03_pipeline.lib.misc.pedigree import Family, Sample
+
+
+def passes_relatedness_check(
+    relatedness_check_lookup: dict[tuple[str, str], list],
+    sample_id: str,
+    other_id: str,
+    relation: Relation,
+) -> bool:
+    # No relationship to check, return true
+    if other_id is None:
+        return True
+    coefficients = relatedness_check_lookup.get(
+        (min(sample_id, other_id), max(sample_id, other_id)),
+    )
+    if not coefficients:
+        return False
+    return np.allclose(
+        coefficients,
+        relation.coefficients,
+        0.1,
+    )
+
+
+def passes_all_relatedness_checks(  # noqa: C901
```

Comment:
To briefly summarize the approach here, the idea is to check every identified relationship in the pedigree for a sample.  A sample succeeds if every relationship is validated.  A family succeeds if every sample succeeds.  If a relationship exists but isn't identifiable in the pedigree (for example a brother/sister pair with no listed parents), we don't have anything to check so the family will succeed.